docs/howto/system_testing.md

```diff
@@ -302,12 +302,11 @@ There are some specific environment variables that should be leveraged to overco
 
 ### Kubernetes service deployer
 
-The Kubernetes service deployer requires the `_dev/deploy/k8s` directory to be present. It can include additional `*.yaml` files to deploy
-custom applications in the Kubernetes cluster (e.g. Nginx deployment). If no resource definitions (`*.yaml` files ) are needed,
-the `_dev/deploy/k8s` directory must contain an `.empty` file (to preserve the `k8s` directory under version control).
+The Kubernetes service deployer requires the `_dev/deploy/k8s` directory to be present. 
+It needs to include either a `kustomization.yaml` file or an `.empty` file. The first is needed to ensure the deployment of each resource,
+while the second is required if no custom resources are needed.
```

Comment:
Not sure if I understand this correctly. Plain resources are not going to be supported anymore, and only kustomize can be used now? This is a breaking change, right?

Reply:
It is a breaking change only for Kubernetes. Which will be immediately fixed. Regarding the `.empty` file: it was already required, so there is no change there. I explained better [in this issue](https://github.com/elastic/elastic-package/issues/1102) why this only affects Kubernetes integration.

Reply:
It is a breaking change in `elastic-package`, that seems that can be prevented without much more effort. I can understand that it probably affects only an integration, but I would really prefer to avoid it.

Accepting the breaking change will complicate updating the integrations repository (elastic-package and the kubernetes integration will need to be updated at the same time, or we need to have CI broken between updates), and it can be a problem if work needs to be done on the kubernetes package with older versions, for backports for example.

Reply:
Btw, following the conversation in the linked issue, it also seems preferred to continue supporting the current method (https://github.com/elastic/elastic-package/issues/1102#issuecomment-1398967317, https://github.com/elastic/elastic-package/issues/1102#issuecomment-1403352269).

Reply:
> elastic-package and the kubernetes integration will need to be updated at the same time, or we need to have CI broken between updates

That is correct. That's why the plan is to open the PR on Kubernetes integration as soon as this one is merged. It probably won't take more than a few hours to correct that.

> it can be a problem if work needs to be done on the kubernetes package with older versions

I thought there was no backport when it comes to integrations?

Reply:
@constanca-m can we enhance the check if kustomisation.yaml exists to use -k, if `.empty`  file exists do nothing otherwise use the -f ? 

So as to keep both manifests and ksutomize?

Reply:
Yes, that is an easy change

Reply:
Well, they are only examples of problems. The main thing is that we prefer to avoid breaking changes, unless they have a very good justification. There are good reasons for that. What I miss here is the justification to introduce this breaking change, when there seem to be preferred and reasonable alternatives.

> I thought there was no backport when it comes to integrations?

We prefer to don't have them, yes, but they are [supported](https://github.com/elastic/integrations/blob/main/docs/developer_workflow_bug_fix_older_package_version.md). The thing is that if someone needs to work with an old integration, even if it is only to try to reproduce an issue, and a breaking change has been introduced, they may find problems and require to downgrade their elastic-package installation.

Reply:
Race condition sending comments :slightly_smiling_face: 

> can we enhance the check if kustomisation.yaml exists to use -k, if `.empty` file exists do nothing otherwise use the -f ?

> Yes, that is an easy change

This looks great to me.

Reply:
That makes sense, thank you for explaining. I agree that `-f` should be supported along with `-k` and I will update to make that possible again. Just adding, for future reference, that `-f` may be necessary if users want to create a new metricset and have no intention in using a kustomization file.